### ISSUE TITLE
Integration of ToniA/arduino-heatpumpir

### ISSUE THREAD

**Issue by grebar** [arendst/Sonoff-Tasmota#3870]
I would like to suggest the integration of @ToniA [arduino-heatpumpir](https://github.com/ToniA/arduino-heatpumpir/) library in the Sonoff-Tasmota firmware.
My suggestion was already proposed by @ripkrizbi in this [post](https://github.com/arendst/Sonoff-Tasmota/issues/3423#issuecomment-416047700).

The library already supports a lot of HVAC and support the introduction of new models in a simle and coerent way.

I already made a preliminary integration in my dev environment, please see this [post](https://github.com/ToniA/arduino-heatpumpir/issues/63#issuecomment-423729181) for further details and open points.

Gregorio.

**Comment by ascillato2** [comment]
> My suggestion was already proposed by @ripkrizbi in this post.

Hi, so if it is already suggested in an open issue with enhacement label to be taken into account, means that this issue is dupplicated. Closing.

**Comment by ascillato** [comment]
Let's continue the suggestions and code proposals at the #3423

**Comment by grebar** [comment]
Here below my follow-up. 
Some days ago I opened a new Issue, referencing this one and proposing just like @ripkrizbi made, the integration of https://github.com/ToniA/arduino-heatpumpir.

After a while (a few hours) @ascillato closed my issue because he tag it as duplicated so no one saw my posted [Issue](https://github.com/arendst/Sonoff-Tasmota/issues/3870).

I tried to integrate the arduino-heatpumpir in Tasmota, and, after while I bought the first problem and first success.

Briefly, I detected a problem that does not allow the integration of arduino-heatpumpir lib in Tasmota, the problem consists in an alreay used definition of the identifiers POWER_ON and POWER_OFF; I described the problem in bullet 1 of this [post](https://github.com/ToniA/arduino-heatpumpir/issues/63#issuecomment-423729181).
I am not very expert of C / C++ languages so I asked for an help on the issue desciped in the linked post.

In the mean while, in my development enviroment, I bypassed the problem described in the post renaming the duplicated identifiers used in arduino-heatpumpir.

So, to intengrate the arduino-heatpumpir in Tasmota, we need to follow the following main steps:
1) rename the POWER_ON/POWER_OFF definitions in arduino-heatpumpir 
File: all files that include POWER_ON / POWER_OFF
rename as specified below
POWER_OFF  -> HVAC_POWER_OFF
POWER_ON  -> HVAC_POWER_ON    

2) copy the arduino-heatpumpir folder in tasmota\lib folder as tasmota\lib\HeatpumpIR

3) apply the changes to xdrv_05_irremote.ino file

I attach my preliminary working version of [xdrv_05_irremote.ino](https://github.com/arendst/Sonoff-Tasmota/files/2425865/xdrv_05_irremote.ino.txt)
.

In my version of xdrv_05_irremote.ino I commented the old unused source code.

In my opinion the next steps for a full integrations are:
1) solve the issue related to duplicated definitions (POWER_ON / POWER_OFF)
2) test and complete my xdrv_05_irremote.ino changes
3) propose the integration in the tasmota to @arendst.

Gregorio
